### PR TITLE
build: :building_construction: terraform improvements and CI

### PR DESCRIPTION
**Pull request by joaquimsn**
- Update Terraforma sintaxe
- Organized terraform file structure
- Added new variable with default value to replace static parameter in main.tf
- deploy megalista metadata to bucket with terraform
- Create Git Action to validate terrafom files

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -56,7 +17,7 @@ locals {
         "jobName": "megalist_daily",
         "parameters": {
             "developer_token": "${var.developer_token}",
-            "client_id": "${var.client_id}",
+            "client_id": "${var.client_id},
```
Comment (caiotomazelli) [inline terraform/main.tf:20]: Was this removed by accident?

File: terraform/main.tf [hunk 2]
```diff
@@ -160,13 +121,13 @@ resource "google_cloud_scheduler_job" "megalista_job" {
...
-  time_zone        = "America/Sao_Paulo"
+  time_zone        = local.time_zone
   attempt_deadline = "320s"
-  region           = "us-central1"
+  region           = var.region
 
   http_target {
     http_method = "POST"
-    uri         = "https://dataflow.googleapis.com/v1b3/projects/${data.google_client_config.current.project}/templates:launch?gcsPath=gs://${var.bucket_name}/templates/megalist"
+    uri         = "https://dataflow.googleapis.com/v1b3/projects/${data.google_client_config.current.project}/templates:launch?gcsPath=gs://${var.bucket_name}/templates/megalista"
```
Comment (caiotomazelli) [inline terraform/main.tf:130]: Template name actually needs to be megalist. See https://github.com/google/megalista/blob/f51d296e11ccee9cebf76d3940cb03513b73e19d/run_cloud.sh#L24

**Review by caiotomazelli — CHANGES_REQUESTED** [review]
Hi Joaquim, thanks for your contribution, much appreciated!

Can you please review the 2 comments below? Also, can you briefly describe how you tested this change?

File: terraform/main.tf (hunk 2 above)
Comment (caiotomazelli) [reply terraform/main.tf:130]: Oops. just saw that you also modifed run_cloud.sh script. I guess it makes sense to keep consistency here. Nevermind this comment.

File: terraform/main.tf (hunk 1 above)
Comment (joaquimsn) [reply terraform/main.tf:20]: Yes, I will adjust

**Comment by joaquimsn** [comment]
@caiotomazelli I performed the test by deploying a new GCP project, and ran the dataflow to create an audience on Google Ads.

**Review by caiotomazelli — APPROVED** [review]

**Pull request merged**